matRad/tools/matRad_recursiveFieldAssignment.m

```diff
@@ -55,38 +56,43 @@
         if ~isfield(assignTo,field)
             assigned.(field) = reference.(field);
         else
-            assigned.(field) = matRad_recursiveFieldAssignment(assignTo.(field),reference.(field),fieldChangedWarningMessage,field);
+            assigned.(field) = matRad_recursiveFieldAssignment(assignTo.(field),reference.(field),overwrite,fieldChangedWarningMessage,field);
         end
     end
 % If the reference is not a struct, we can assign it directly. 
 % However, we need to check if the assignTo is a struct and if we need to warn the user of overwriting a struct with a non-struct
 elseif isstruct(assignTo) && ~isstruct(reference)        
-    if ~isempty(fieldname)
-        matRad_cfg = MatRad_Config.instance();
-        matRad_cfg.dispWarning([fieldChangedWarningMessage 'Field ''%s'' is a struct but will be overwritten by a ''%s!'''],fieldname,class(reference));
+    if overwrite
+        if ~isempty(fieldname)
+            matRad_cfg = MatRad_Config.instance();
+            matRad_cfg.dispWarning([fieldChangedWarningMessage 'Field ''%s'' is a struct but will be overwritten by a ''%s!'''],fieldname,class(reference));
+        end
+        assigned = reference;
+    else
+        assigned = assignTo;
     end
-    assigned = reference;
 % If the assignTo is not a struct, we can assign it directly.
 % However, we need to check if the reference is a struct and if we need to warn the user of overwriting a non-struct with a struct
 elseif ~isstruct(assignTo) && isstruct(reference)
-    if ~isempty(fieldname)
-        matRad_cfg = MatRad_Config.instance();
-        matRad_cfg.dispWarning([fieldChangedWarningMessage 'Field ''%s'' is not a struct but will be overwritten by a struct!'],fieldname);
+    
+    if overwrite
+        if ~isempty(fieldname)
+            matRad_cfg = MatRad_Config.instance();
+            matRad_cfg.dispWarning([fieldChangedWarningMessage 'Field ''%s'' is not a struct but will be overwritten by a struct!'],fieldname);
+        end
+        assigned = reference;
+    else
+        assigned = assignTo;
     end
-    assigned = reference;
 else
-    if ~isempty(fieldname) && ~isempty(fieldChangedWarningMessage)
-        matRad_cfg = MatRad_Config.instance();
-        if isequal(class(assignTo),class(reference))
-            matRad_cfg.dispWarning([fieldChangedWarningMessage 'Field ''%s'' will be overwritten by reference value!'],fieldname);
-        else
+    if overwrite
+        if ~isempty(fieldname) && ~isempty(fieldChangedWarningMessage)
+            matRad_cfg = MatRad_Config.instance();
             matRad_cfg.dispWarning([fieldChangedWarningMessage 'Field ''%s'' is supposed to be a %s but will be overwritten by a ''%s!'''],fieldname,class(assignTo),class(reference));
```

Comment:
I think that this warning should only be issued if the classes are different, i.e. we need a 
`if ~isequal(class(assignTo),class(reference))` here?